packages/run-protocol/src/vpool-xyk-amm/swap.js

```diff
@@ -19,23 +27,36 @@ export const makeMakeSwapInvitation = (zcf, provideVPool) => {
       give: { In: amountIn },
       want: { Out: amountOut },
     } = seat.getProposal();
-    const pool = provideVPool(amountIn.brand, amountOut.brand).externalFacet;
-    return pool.swapIn(seat, amountIn, amountOut);
+    if (args) {
+      AmountMath.coerce(amountOut.brand, args.maxOut);
+      // TODO check that there are no other keys
+    }
+
+    const pool = provideVPool(amountIn.brand, amountOut.brand).internalFacet;
+    let prices;
+    const maxOut = args && args.maxOut;
```

Comment:
turadg and I have started using
```suggestion
    const maxOut = args?.maxOut;
```

Reply:
They are equivalent, and the current code is clearer to me. Also, the type verifier doesn't treat them the sameyet and some sometimes that complains.

Reply:
### ### So feel free to change, but out of scope for this PR :)